### PR TITLE
util/docker-container: create docker container

### PR DESCRIPTION
**Pull request by ALEX11BR**
adds a dockerfile for this course's docker container

# Prerequisite Checklist

- [x] Read the [contribution guidelines](https://github.com/open-education-hub/ccas-internal/blob/main/CONTRIBUTING.md#pull-requests) regarding submitting new changes to the project;
- [ ] Tested your changes against relevant architectures and platforms;
- [ ] Updated relevant documentation (if needed).

### CODE REVIEW

File: util/docker-container/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+RUN apt-get update -y
+RUN apt-get install -y gcc gcc-multilib nasm make build-essential
+RUN apt-get install -y binutils
+RUN apt-get install -y gdb
+RUN apt-get install -y vim nano
+RUN apt-get install -y tmate
+RUN apt-get install -y python2.7
+RUN apt-get install -y python3.10 python-is-python3
+
+CMD [ "bash" ]
```
Comment (gabrielmocanu) [inline util/docker-container/Dockerfile]:
```suggestion
RUN apt-get update \
    && apt-get install -y \
        gcc \
        gcc-multilib \
        nasm \
        make \
        build-essential \
        binutils \
        gdb \
        vim \
        nano \
        tmate \
        python2.7 \
        python3.10 \
        python-is-python3 \
    && apt-get clean \
    && rm -rf /var/lib/apt/lists/*

CMD [ "bash" ]
```

Comment (gabrielmocanu) [reply util/docker-container/Dockerfile]: [Here] are some best practices about Dockerfile. We need to put all these commands inside a single one in order to have fewer layers.

[Here]: https://docs.docker.com/develop/develop-images/instructions/

**Review by gabrielmocanu — CHANGES_REQUESTED** [review]
I will add here some other things:
1. Please move this intro a directory called `hsi-runner` instead of `docker-container`.
2. Keep the same structure as [here].
3. You can copy the README file from there and change only URLs
4. Add a Makefile, similar to [here]

[here]: https://github.com/open-education-hub/operating-systems/tree/main/util/os-runner

**Comment by teodutu** [comment]
Why are you adding this here? It will generate merge conflicts and not be available for the cs-pub-ro repo [1] until we sync them. I think it's best to do the development on our fork [1] and then periodically sync it [1] with this one via a PR bot such as [2].

[1] https://github.com/cs-pub-ro/hardware-software-interface/
[2] https://github.com/open-education-hub/operating-systems/pull/350

**Comment by gabrielmocanu** [comment]
OK, @ALEX11BR , I think @teodutu  is right, sorry for confusion.
Please close this PR and open the same PR but on this repo: https://github.com/cs-pub-ro/hardware-software-interface/

**Pull request closed without merging**